packages/@aws-cdk/aws-appmesh/lib/shared-interfaces.ts

```diff
@@ -92,9 +92,65 @@ export interface VirtualNodeListener {
   readonly portMapping?: PortMapping;
 
   /**
-   * Array fo HealthCheckProps for the node(s)
+   * Health checking strategy upstream nodes should use when communicating with the listener
    *
    * @default - no healthcheck
    */
   readonly healthCheck?: HealthCheck;
 }
+
+/**
+ * All Properties for Envoy Access logs for mesh endpoints
+ */
+export interface AccessLogConfig {
+  /**
+   * Path to a file to write access logs to
+   *
+   * @default - no file based access logging
+   */
+  readonly filePath?: string;
+}
+
+/**
+ * Configuration for Envoy Access logs for mesh endpoints
+ */
+export abstract class AccessLog {
+  /**
+   * Path to a file to write access logs to
+   *
+   * @default - no file based access logging
+   */
+  public static fromFilePath(filePath: string): AccessLog {
+    return new FileAccessLog(filePath);
+  }
+
+  /**
+   * Called when the AccessLog type is initialized. Can be used to enforce
+   * mutual exclusivity with future properties
+   *
+   */
+  public abstract bind(scope: cdk.Construct): AccessLogConfig;
+}
+
+/**
+ * Configuration for Envoy Access logs for mesh endpoints
+ */
+export class FileAccessLog extends AccessLog {
```

Comment:
Let's actually _not_ export this file, and keep it module-private (just remove the `export` keyword).